NPC Base interaction

MGHMSJ5 opened the pull request:
Added example for NPC interaction.

IMPORTANT: When interacting with the NPC, you need to manually disable the 'dialogue' object in the inspector for it to work. This will later change when the dialogue is properly implemented.

Important changes:
BaseInteract.cs (fixed the single interaction)
Interacting.unity (Scene)

Added:
NPCInteractEXAMPLE.cs (script to be able to interact with the NPC)
NPCInteractEXAMPLE.prefab
(white material)

MGHMSJ5 committed:
Added 'interactOnce'

feat(interact): made it so that all BaseInteract has the option to only have the player be able to interact with it once

MGHMSJ5 committed:
New Material

feat(mat): white material

MGHMSJ5 committed:
Added example for NPC interaction

feat(interact): NPC interaction.
+ script
+ prefab
changed the Interacting scene by adding the NPC example to it

fix(interact): fixed the interact once problem.

MGHMSJ5 committed:
Interact text change

refactor(ui): edited the Canvas UI by changing the text of the interactions to be the buttons

MGHMSJ5 committed:
Variable name

refactor(interact): changed variable name in BaseInteract for consistency

MGHMSJ5 committed:
Added comments

refactor(interact): added comments to BaseInteract and NPCInteractEXAMPLE

MGHMSJ5 committed:
Interact fix

fix(interact): moved some code to fix trigger issue

NickRisse commented:
Hey @MGHMSJ5, while reviewing your PR, I'd suggest the following code changes:

👉 [**Code Suggestion for #9**](https://gitkraken.dev/drafts/b06bf63d-69af-403d-a96f-19a5c5b1a5f6?source=PRComment)
>Added two comments for clarification and a design question.

https://github.com/MGHMSJ5/Project-Capstone/pull/9

You can also review and apply these suggestions [locally on your machine](https://gitkraken.dev/link/drafts/b06bf63d-69af-403d-a96f-19a5c5b1a5f6?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPT1BsZ2I4NlJ4NExYIl0=&source=PRComment).

<details><summary>Learn more about GitKraken Code Suggest</summary>

>[Code Suggest](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbS9zb2x1dGlvbnMvY29kZS1zdWdnZXN0P3NvdXJjZT1naXRodWIuY29tJm1lZGl1bT1wcl9jb21tZW50?source=PRSuggest&event=redirectToCodeSuggestPage&provider=github) liberates your code reviews from GitHub's restrictive, comment-only feedback style. As simple as suggesting changes in a Google-doc, provide real code suggestions from where you code, e.g. your IDE, and on anything in your project — not just on the lines of code changed in the PR.

_Join your team on [GitKraken](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbT9zb3VyY2U9Z2l0aHViLmNvbSZtZWRpdW09cHJfY29tbWVudA==?source=PRSuggest&event=redirectToIndexPage&provider=github) to speed up PR review._
</details>

MGHMSJ5 committed:
Suggested change on code comments

MGHMSJ5 committed:
NPC dialogue options

feat(NPC interact): changed the script so that 2 dialogue options are available.

NickRisse commented:
Hey @MGHMSJ5, while reviewing your PR, I'd suggest the following code changes:

👉 [**Code Suggestion for #9**](https://gitkraken.dev/drafts/04bc4ef2-ff24-4d3d-b2ea-1fdfd5c6e8ea?source=PRComment)
>Fixed grammar errors. Rest looks very solid now.

https://github.com/MGHMSJ5/Project-Capstone/pull/9

You can also review and apply these suggestions [locally on your machine](https://gitkraken.dev/link/drafts/04bc4ef2-ff24-4d3d-b2ea-1fdfd5c6e8ea?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPT1BsZ2I4NlJ4NExYIl0=&source=PRComment).

<details><summary>Learn more about GitKraken Code Suggest</summary>

>[Code Suggest](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbS9zb2x1dGlvbnMvY29kZS1zdWdnZXN0P3NvdXJjZT1naXRodWIuY29tJm1lZGl1bT1wcl9jb21tZW50?source=PRSuggest&event=redirectToCodeSuggestPage&provider=github) liberates your code reviews from GitHub's restrictive, comment-only feedback style. As simple as suggesting changes in a Google-doc, provide real code suggestions from where you code, e.g. your IDE, and on anything in your project — not just on the lines of code changed in the PR.

_Join your team on [GitKraken](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbT9zb3VyY2U9Z2l0aHViLmNvbSZtZWRpdW09cHJfY29tbWVudA==?source=PRSuggest&event=redirectToIndexPage&provider=github) to speed up PR review._
</details>

MGHMSJ5 committed:
Nick Feedbacl

MGHMSJ5 committed:
Feedback Iteraction

refactor(NPC interact): removed the weird lines and fixed the dialogie to dialogue

NickRisse reviewed (approved):
Looks dope!
